src/expr/impl/src/scalar/array.rs

```diff
@@ -14,16 +14,20 @@
 
 use risingwave_common::array::{ListValue, StructValue};
 use risingwave_common::row::Row;
-use risingwave_common::types::{DataType, ListRef, MapType, MapValue, ToOwnedDatum};
+use risingwave_common::types::{
+    DataType, ListRef, MapRef, MapType, MapValue, ScalarRefImpl, ToOwnedDatum,
+};
 use risingwave_expr::expr::Context;
 use risingwave_expr::{function, ExprError};
 
-#[function("array(...) -> anyarray", type_infer = "panic")]
+use super::array_positions::array_position;
+
+#[function("array(...) -> anyarray", type_infer = "unreachable")]
```

Comment:
What's the different between type_infer = "panic" and "unreachable"?

Reply:
I renamed `panic` to `unreachable` in this PR (`src/expr/macro/src/gen.rs`). You can check the panic message there.